File: VersionC++/player.cpp
#include <string>

#include "player.h"
using namespace std;



Player::Player(int boardSize)
{
    int i;
    int possibleBoardSizes[] = { 3, 4, 5, 6, 8 };
    int numberOfStones[] = { 10, 15, 21, 30, 50 };
    
    for(i = 0; i < 5 && boardSize != possibleBoardSizes[i]; i++);
    
    if(i != 5)
    {
        stonesLeft = numberOfStones[i];
        capsotnesLeft = i / 2;
    }
}

string Player::getName() { return mName; }
void Player::setName(string pName) {mName = pName;}
int Player::getStonesLeft(){ return stonesLeft; }
int Player::getCapstonesLeft(){ return capsotnesLeft; }


void Player::decrementStones() { stonesLeft--; }
bool Player::takeCapstone() 
{ 
    if(capsotnesLeft)
    {
        capsotnesLeft--;
        return true;
    }
    return false;
}

File: Unity Version/Assets/Scripts/Player.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Player{
    
    private string playerName;
    private int stonesLeft;
    private int capstonesLeft;

    private Dictionary<int, int> stoneNumbers = new Dictionary<int, int>
    {
        { 3, 10 },
        { 4, 15 },
        { 5, 21 },
        { 6, 30 },
        { 8, 50 }
    };

    public Player(string pName, int boardSize)
    {
        playerName = pName;
        stonesLeft = stoneNumbers[boardSize];
        capstonesLeft = (boardSize - 3) / 2;
    }

    public bool TakeStone()
    {
        return (--stonesLeft > 0);
    }

    public bool HasCapstone()
    {
        return capstonesLeft > 0;
    }

    public void TakeCapstone()
    {
        capstonesLeft--;
    }

    public int GetStonesLeft()
    {
        return stonesLeft;
    }

    public int GetCapstonesLeft()
    {
        return capstonesLeft;
    }

    public string GetName()
    {
        return playerName;
    }
}
